ext/intl/uchar/uchar.stub.php

```diff
@@ -0,0 +1,338 @@
+<?php
+
+class IntlChar
+{
+    /**
+     * @param mixed $codepoint
+     * @return bool|null
+     */
+    public static function hasBinaryProperty($codepoint, int $property) {}
+
+    /**
+     * @param mixed $codepoint
+     * @return array|null
+     */
+    public static function charAge($codepoint) {}
+
+    /**
+     * @param mixed $codepoint
+     * @return int|null
+     */
+    public static function charDigitValue($codepoint) {}
+
+    /**
+     * @param mixed $codepoint
+     * @return int|null
+     */
+    public static function charDirection($codepoint) {}
+
+    /** @return int|null */
+    public static function charFromName(string $characterName, int $nameChoice = IntlChar::UNICODE_CHAR_NAME) {}
+
+    /**
+     * @param mixed $codepoint
+     * @return int|string|null
+     */
+    public static function charMirror($codepoint) {}
+
+    /**
+     * @param mixed $codepoint
+     * @return string|null
+     */
+    public static function charName($codepoint, int $nameChoice = IntlChar::UNICODE_CHAR_NAME) {}
+
+    /**
+     * @param mixed $codepoint
+     * @return int|null
+     */
+    public static function charType($codepoint) {}
+
+    /**
+     * @param mixed $codepoint
+     * @return string|null
+     */
+    public static function chr($codepoint) {}
+
+    /**
+     * @param mixed $codepoint
+     * @return int|false|null
+     */
+    public static function digit($codepoint, int $radix = 10) {}
+
+    /**
+     * @param mixed $start
+     * @param mixed $limit
+     * @param callable $callback
+     * @return bool|null
+     */
+    public static function enumCharNames($start, $limit, $callback, int $nameChoice = IntlChar::UNICODE_CHAR_NAME) {}
+
+    /**
+     * @param callable $callable
+     * @return void
+     */
+    public static function enumCharTypes($callable) {}
```

Comment:
As this uses `f` in zpp, you should be able to include a proper `callable` type, not just doc comment.

Reply:
There is the open issue about callable not working correctly, but as this is in a method, it existing zpp check is also already broken, so it ends up not mattering :)

Reply:
Yeah, that issue is the reason why I didn't use a type declaration. I remember there were failures in Zend tests, but as it turns out using `callable` type is not a problem unless it breaks a test :)